File: assignment_3.py
def maxProduct(nums):
    max_nums=nums[0]*nums[1]
    for i in range(0,len(nums)):
        for j in range(i,len(nums)-1):
            if max_nums < nums[i]*nums[j+1]:
                max_nums = nums[i]*nums[j+1]
            j+=1
        i+=1 
    print(f"得到{max_nums}為最大值")   

maxProduct([5, 20, 2, 10, 6, 10])
maxProduct([5, 30, 2, 4, 6, 10])
maxProduct([-1, 2, -10, 10, 20, 20])
maxProduct([-1, 2, -10, -10, -20, 20])
maxProduct([-1, 2, 20, 20, -20, -20])
maxProduct([-1,20])




    

File: assignment_3.js
function maxProduct(nums){
    var a = nums[0];
    var b = nums[0];
    var c = nums[0];
    var d = nums[0];
    for(var i = 0; i <=nums.length; i++){
        if(nums[i] > a && nums[i] > b){
            b = a;
            a = nums[i];
        }
        else if (nums[i] > b){
            b = nums[i];
        } 
    }
    for(var j = 0; j <=nums.length; j++){
        if(nums[j] < c && nums[j] < d){
            d = c;
            c = nums[j];
        }
        else if (nums[j] < d){
            d = nums[j];
        } 
    }
    var pos_max=a*b;
    var neg_max=c*d;
    if(pos_max > neg_max){
        console.log(`得到${pos_max} 因為 ${a} 和 ${b} 相乘得到最大值`);
        }
    else if(pos_max < neg_max){
        console.log(`得到${neg_max} 因為 ${c} 和 ${d} 相乘得到最大值`);
        }
    else{
        console.log(`得到${pos_max} 因為 ${a} 和 ${b} 還有${c} 和 ${d} 相乘得到最大值`);
        }
}
maxProduct([5, 20, 2, 20, 6, 10])
maxProduct([5, 20, 2, 4, 6, 10])
maxProduct([-1, 2, -10, 10, 20, 20])
maxProduct([-1, 2, -10, -10, -20, 20])
maxProduct([-1, 2, 20, 20, -20, -20])
